docs/reference/esql/functions/kibana/definition/bucket.json

```diff
@@ -13,21 +13,39 @@
           "description" : ""
         },
         {
-          "name" : "buckets",
+          "name" : "bucketsOrSpan",
+          "type" : "date_period",
+          "optional" : false,
+          "description" : ""
+        }
+      ],
+      "variadic" : false,
+      "returnType" : "datetime"
+    },
+    {
+      "params" : [
+        {
+          "name" : "field",
+          "type" : "datetime",
+          "optional" : false,
+          "description" : ""
+        },
+        {
+          "name" : "bucketsOrSpan",
           "type" : "integer",
           "optional" : false,
           "description" : ""
         },
         {
           "name" : "from",
           "type" : "datetime",
-          "optional" : false,
+          "optional" : true,
           "description" : ""
         },
         {
           "name" : "to",
           "type" : "datetime",
-          "optional" : false,
+          "optional" : true,
```

Comment:
I think this is not correct; in the `integer` case, neither `from` nor `to` are optional, no?

Applies below as well.

Reply:
Yeh, not sure how to go about this file, it being generated: the optionality of the parameters can't be accurately specified for functions with more than one optional parameter (the two or four bit), or based on types of various non-optional parameters.

Reply:
Oh, that's not ideal. That means we probably want to update our docs generation a bit. (out of scope, in this case)

CC @nik9000 , if you agree I'll open an issue so we don't forget.